dspace-api/src/main/java/org/dspace/app/profile/ResearcherProfileServiceImpl.java

```diff
@@ -0,0 +1,322 @@
+        }
+
+        context.turnOffAuthorisationSystem();
+        itemService.addMetadata(context, item, "dspace", "object", "owner", null,
+                                ePerson.getName(), ePerson.getID().toString(), CF_ACCEPTED);
+        context.restoreAuthSystemState();
+
+        return new ResearcherProfile(item);
+    }
+
+    @Override
+    public boolean hasProfileType(Item item) {
+        String profileType = getProfileType();
+        if (StringUtils.isBlank(profileType)) {
+            return false;
+        }
+        return profileType.equals(itemService.getEntityType(item));
+    }
+
+    @Override
+    public String getProfileType() {
+        return configurationService.getProperty("researcher-profile.entity-type", "Person");
+    }
+
+    private Optional<Item> findItemByURI(final Context context, final URI uri) throws SQLException {
+        String path = uri.getPath();
+        UUID uuid = UUIDUtils.fromString(path.substring(path.lastIndexOf("/") + 1));
+        return ofNullable(itemService.find(context, uuid));
+    }
+
+    /**
+     * Search for an profile item owned by an eperson with the given id.
+     */
+    private Item findResearcherProfileItemById(Context context, UUID id) throws SQLException, AuthorizeException {
+
+        String profileType = getProfileType();
+
+        Iterator<Item> items = itemService.findByAuthorityValue(context, "dspace", "object", "owner", id.toString());
+        while (items.hasNext()) {
+            Item item = items.next();
+            String entityType = itemService.getEntityType(item);
+            if (profileType.equals(entityType)) {
+                return item;
+            }
+        }
+
+        return null;
+    }
+
+    /**
+     * Returns a Profile collection based on a configuration or searching for a
+     * collection of researcher profile type.
+     */
+    @SuppressWarnings("rawtypes")
+    private Optional<Collection> findProfileCollection(Context context) throws SQLException, SearchServiceException {
+        UUID uuid = UUIDUtils.fromString(configurationService.getProperty("researcher-profile.collection.uuid"));
+        if (uuid != null) {
+            return ofNullable(collectionService.find(context, uuid));
+        }
+
+        String profileType = getProfileType();
+
+        DiscoverQuery discoverQuery = new DiscoverQuery();
+        discoverQuery.setDSpaceObjectFilter(IndexableCollection.TYPE);
+        discoverQuery.addFilterQueries("dspace.entity.type:" + profileType);
+
+        DiscoverResult discoverResult = searchService.search(context, discoverQuery);
+        List<IndexableObject> indexableObjects = discoverResult.getIndexableObjects();
+
+        if (CollectionUtils.isEmpty(indexableObjects)) {
+            return empty();
+        }
+
+        if (indexableObjects.size() > 1) {
+            log.warn("Multiple " + profileType + " type collections were found during profile creation");
+            return empty();
```

Comment:
@LucaGiamminonni : I think we need to rethink this.  In testing this today, I immediately hit strange behavior because of this undocumented requirement.  

If I run this PR with the frontend PR, I get the "Create New" button for a Profile, but if I click it, I get a very confusing HTTP error:
![profile](https://user-images.githubusercontent.com/483997/166809904-497c5b77-98bb-4115-a4a9-fa3af0751645.png)

The cause of the error is that I seem to have multiple Collections configured accepting `Person` entities.  This default setup is going to happen **frequently**, so having Researcher Profiles throw an error by default is not a good behavior.

I'd recommend we either find a way to remove the error that requires sites to have a single Person Collection to create new Researcher Profiles .. or, alternatively, we **require `researcher-profile.collection.uuid` to be set** in order to enable Researcher Profiles.  If we choose to require it, I think we should either throw a more useful error message in the UI, or remove the `Researcher Profile` section if it's not configured.

(Obviously our Submission process already supports the ability to create Person entities in multiple Collections. So, it's odd that Researcher Profiles (which are just Person Entities) needs to behave differently)